### PR TITLE
Social embeds end to end test

### PR DESCRIPTION
**Pull request by noagb**
Resolves #NUMBER

**Overall change:**
_End to end test for optimo Article Social embeds_

**Code changes:**

- _Added a social embed test for each social media for testsThatFollowSmokeTestConfig in test.js_
- _Test were added in order to execute against cannonical and AMP pages_
- _Test were added to test rendering of embeded social posts_

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project
- [ ] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: cypress/integration/pages/articles/tests.js [hunk 1]
```diff
@@ -184,11 +185,132 @@ export const testsThatFollowSmokeTestConfig = ({
           });
         });
       });
+
+      describe.only('Social Embeds', () => {
+        it('Youtube embed is redered when it exists on page', () => {
+          cy.request(`${Cypress.env('currentPath')}.json`).then(({ body }) => {
+            const youtubeSocialEmbedsData = body.content.model.blocks.filter(content => content.type === 'social')
```
Comment (amoore108) [inline cypress/integration/pages/articles/tests.js]: You may be able to use the helper `getBlockData` and `getBlockType` functions for extracting the block data.

File: cypress/integration/pages/articles/tests.js [hunk 2]
```diff
@@ -184,11 +185,132 @@ export const testsThatFollowSmokeTestConfig = ({
...
+              .filter(content => content.model.providerName === 'YouTube');
+            // cy.log(JSON.stringify(youtubeSocialEmbedsData));
```
Comment (amoore108) [inline cypress/integration/pages/articles/tests.js]: Couple of commented bits here that could be removed.

File: cypress/integration/pages/articles/tests.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
+// import { onlyOn, skipOn } from '@cypress/skip-test';
```
Comment (amoore108) [inline cypress/integration/pages/articles/tests.js]: Can remove this if its not used.

**Comment by amoore108** [comment]
Looks good. Theres a few linting errors, so you may need to enable ESLint on save in your code editor to ensure that the ESLint rules are being automatically applied.

File: cypress/integration/pages/articles/tests.js [hunk 4]
```diff
@@ -184,11 +185,132 @@ export const testsThatFollowSmokeTestConfig = ({
...
+                cy.get(`[data-e2e="instagram-embed-${instagramUrl}"]`)
+                  .scrollIntoView()
+                  .within(() => {
+                    cy.get(`[href^="#end-of-instagram-content"]`).should('exist');
+                    cy.get(`iframe`).should('exist');
+                  });
+              });
+            } else {
+              // cy.log('There is no Instagram embed in this page!');
+              // this.skip();
```
Comment (david-boydell) [inline cypress/integration/pages/articles/tests.js]: Do we need these comments? If not I'd remove, same for l. 258 & 259

File: cypress/integration/pages/articles/tests.js [hunk 5]
```diff
@@ -184,11 +185,132 @@ export const testsThatFollowSmokeTestConfig = ({
           });
         });
       });
+
+      describe.only('Social Embeds', () => {
```
Comment (david-boydell) [inline cypress/integration/pages/articles/tests.js]: Is this `only` intended? If not then we should get rid.

File: cypress/integration/pages/articles/tests.js (hunk 1 above)
Comment (noagb) [reply cypress/integration/pages/articles/tests.js]: Seems like those function (`getBlockType`) only return the first occurrence of that block type. 
I will implement new function that return multiple social blocks and select by Provider name.

**Review by amoore108 — APPROVED** [review]

**Review by david-boydell — APPROVED** [review]

**Review by david-boydell — APPROVED** [review]

**Review by david-boydell — APPROVED** [review]

**Pull request merged**